### PR TITLE
add runInTransaction and wrapInTransaction

### PR DESCRIPTION
**Pull request by lytc**
Hi @odavid , I created this PR for this request https://github.com/odavid/typeorm-transactional-cls-hooked/issues/92, could you help to have a look?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
   },
   "dependencies": {
     "@types/cls-hooked": "^4.2.1",
-    "cls-hooked": "^4.2.2"
+    "cls-hooked": "^4.2.2",
+    "delay": "^5.0.0"
```
Comment (odavid) [inline package.json:42]: Please change this dependency to be a dev dependency

File: src/wrapInTransaction.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            return runOriginal()
+          } else {
+            return runWithNewHook()
+          }
+      }
+    }) as unknown as ReturnType<Func>;
+  }
+
+  return wrapped as Func;
+}
```
Comment (odavid) [inline src/wrapInTransaction.ts:133]: Please [add a newline](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file) here...

**Review by odavid — CHANGES_REQUESTED** [review]

**Comment by odavid** [comment]
Hi @lytc, thanks for the contribution!

I feel you also need to add the `runInTransaction` function to the exports within the index.ts

**Comment by koenpunt** [comment]
Not sure if this is the correct thread to bring this up, but I feel it might be related; I'm using this library indirectly (through [typeorm-test-transactions](https://www.npmjs.com/package/typeorm-test-transactions)), and I was wondering if it would possible to have separate "start" and "stop" transaction methods? So not a function wrapper (or decorator).

I hope to be able to use it in before/after hooks of jest, to automatically "wrap" every test in a transaction;

```ts
beforeEach(() => {
  startTransaction();
});

test('do something with the database', async () => {
    await someDbAction();
});

afterEach(() => {
  rollbackTransaction();
});
```

**Comment by lytc** [comment]
> Hi @lytc, thanks for the contribution!
> 
> I feel you also need to add the `runInTransaction` function to the exports within the index.ts

**Comment by lytc** [comment]
Hi @odavid ,
Thanks for your review, I make the change regarding to your comments!

**Comment by odavid** [comment]
Thank you!

**Pull request merged**